setup.py

```diff
@@ -8,55 +8,98 @@
+     ```
+     git checkout main
+     git pull upstream main
+     git checkout -b release
+     ```
+
+2. Change the version to the release VERSION in:
    - __init__.py
    - setup.py
 
-2. Commit these changes: "git commit -m 'Release: VERSION'"
-
-3. Add a tag in git to mark the release: "git tag VERSION -m 'Add tag VERSION for pypi'"
-   Push the tag to remote: git push --tags origin main
+3. Commit these changes, push and create a Pull Request:
+     ```
+     git add -u
+     git commit -m "Release: VERSION"
+     git push upstream release
+     ```
+   - Go to: https://github.com/huggingface/datasets/pull/new/release
+   - Create pull request
 
-4. Build both the sources and the wheel. Do not change anything in setup.py between
+4. From your local release branch, build both the sources and the wheel. Do not change anything in setup.py between
    creating the wheel and the source distribution (obviously).
+   - First, delete any building directories that may exist from previous builds:
+     - build
+     - dist
+   - From the top level directory, build the wheel and the sources:
+       ```
+       python setup.py bdist_wheel
+       python setup.py sdist
+       ```
+   - You should now have a /dist directory with both .whl and .tar.gz source versions.
 
-   First, delete any "build" directory that may exist from previous builds.
-
-   For the wheel, run: "python setup.py bdist_wheel" in the top level directory.
-   (this will build a wheel for the python version you use to build it).
-
-   For the sources, run: "python setup.py sdist"
-   You should now have a /dist directory with both .whl and .tar.gz source versions.
-
-5. Check that everything looks correct by uploading the package to the pypi test server:
-
-   twine upload dist/* -r pypitest --repository-url=https://test.pypi.org/legacy/
-
+5. Check that everything looks correct by uploading the package to the test PyPI server:
+     ```
+     twine upload dist/* -r pypitest --repository-url=https://test.pypi.org/legacy/
+     ```
    Check that you can install it in a virtualenv/notebook by running:
-   pip install huggingface_hub fsspec aiohttp
-   pip install -U tqdm
-   pip install -i https://testpypi.python.org/pypi datasets
+     ```
+     pip install huggingface_hub fsspec aiohttp
+     pip install -U tqdm
+     pip install -i https://testpypi.python.org/pypi datasets
+     ```
 
-6. Upload the final version to actual pypi:
-   twine upload dist/* -r pypi
+6. Upload the final version to the actual PyPI:
+     ```
+     twine upload dist/* -r pypi
+     ```
 
-7. Fill release notes in the tag in github once everything is looking hunky-dory.
+7. Make the release on GitHub once everything is looking hunky-dory:
+   - Merge the release Pull Request
+   - Create a new release: https://github.com/huggingface/datasets/releases/new
+   - Choose a tag: Introduce the new VERSION as tag, that will be created when you publish the release
+     - Create new tag VERSION on publish
```

Comment:
I'm wondering if it will trigger the release-conda CI job - it's configured like
```yaml
on:
  push:
    tags:
      - "[0-9]+.[0-9]+.[0-9]+*"
```

Reply:
I guess releated to:
- #5251

Reply:
@lhoestq maybe we could align the triggering of release-conda to be the same as the one for generating the docs? That is: push to branch `v*-release`

Reply:
Indeed, conda-release was not triggered for 2.7.0 release...

Reply:
We maintain the huggingface channel but not the conda forge one, which is maintained by the community

Reply:
OK!

Reply:
I have reverted: 8694901

Reply:
Push tag for both sounds good :)